drafts/draft-brockners-ippm-ioam-data-integrity.xml

```diff
@@ -484,51 +484,180 @@
     </section>
 
     <section title="Methods of providing integrity to IOAM data fields">
-      <t>This section describes enhancements to IOAM Trace-Option header to
-      carry a "Trace Signature" field. This can be achieved using symmetric or
+      <t>This section describes enhancements to IOAM Options to carry a
+      intergrity data fields. This can be achieved using symmetric or
       asymetric key based signatures as described below sub-sections.</t>
 
-      <figure>
-        <artwork>
+      <section anchor="IP_Options" title="Integrity Protected IOAM Options">
+        <t>Each of the IOAM Options defined in <xref
+        target="I-D.ietf-ippm-ioam-data"/> are extended to include Integrity
+        Protected(IP) options by allocating the following IOAM Option-Type in
+        the IOAM Option-Type registry.</t>
+
+        <t><list style="hanging">
+            <t hangText="64">IOAM Pre-allocated Trace Intergrity Protected
+            Option-Type corresponds to IOAM Pre-allocated Trace Option-Type
+            with integrity protection.</t>
+
+            <t hangText="65">IOAM Incremental Trace Intergrity Protected
+            Option-Type corresponds to IOAM Incremental Trace Option-Type with
+            integrity protection.</t>
+
+            <t hangText="66">IOAM POT Intergrity Protected Option-Type
+            corresponds to IOAM POT Option-Type with integrity protection.</t>
+
+            <t hangText="67">IOAM E2E Intergrity Protected Option-Type
+            corresponds to IOAM E2E Option-Type with integrity protection.</t>
+          </list></t>
+      </section>
+
+      <section anchor="IP_Sub-header"
+               title="Integrity Protected IOAM Options sub-header">
+        <t>The integrity data sub-header that will be included in IOAM
+        Integrity Protected Options is defined as follows:</t>
+
+        <t><figure>
+            <artwork>
+
+     0                   1                   2                   3
+    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |Signature-suite|  Nonce length |         Reserved.             |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                        Nonce                                  ~
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                       Signature                               ~
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+
+</artwork>
+          </figure></t>
+
+        <t><list style="hanging">
+            <t hangText="Signature-suite:">8-bit defines the algorithms used
+            to compute digest and signature over the Option header and data
+            excluding the Signature field.</t>
```

Comment:
> 8-bit defines the algorithms used to compute digest and signature over the Option header and data excluding the Signature field.

8-bit unsigned integer. This field defines the algorithms used to compute digest and signature over the Option header and data excluding the Signature field.